ci/bazel-scripts/diff.sh

```diff
@@ -13,7 +13,11 @@ set -euo pipefail
 set -x
 cd "$(git rev-parse --show-toplevel)"
 
-MERGE_BASE="${MERGE_BASE_SHA:-HEAD}"
+# since default branch is not the same in ic-private, we can't set it to master, throw error if missing
+git fetch origin $DEFAULT_BRANCH
+TARGET_BRANCH_NAME="${TARGET_BRANCH_NAME:-$DEFAULT_BRANCH}"
+TARGET_BRANCH_SHA=$(git rev-parse origin/$TARGET_BRANCH_NAME)
+MERGE_BASE=$(git merge-base HEAD "$TARGET_BRANCH_SHA")
```

Comment:
Wouldn't this cause the problem we discussed yesterday where `HEAD` on PRs points to a merge between the PR branch and the target branch meaning that this `git merge-base HEAD "$TARGET_BRANCH_SHA"` will return that merge commit which then results in `COMMIT_RANGE` being `HEAD..` which then causes `DIFF_FILES` to become empty?

I'm doing these git calculations in my head so I could be wrong here.

Reply:
Oh I realise I'm wrong. If HEAD points to a merge of the PR with the target branch `git merge-base HEAD "$TARGET_BRANCH_SHA"` should result in `$TARGET_BRANCH_SHA`.
Meaning `COMMIT_RANGE=$TARGET_BRANCH_SHA..`.  So that could actually work.

Reply:
Yeah I double checked this and from my understanding `git merge-base` will always return the common ancestor (so not the merge commit). It's the same as `MERGE_BASE_SHA: ${{ github.event.pull_request.base.sha }}`. The one that could return a merge commit would be the latest commit, which is `CI_COMMIT_SHA: ${{ github.sha }}`.